source/common/network/win32_socket_handle_impl.h

```diff
@@ -42,8 +43,37 @@ class Win32SocketHandleImpl : public IoSocketHandleImpl {
                                    RecvMsgOutput& output) override;
   Api::IoCallUint64Result recv(void* buffer, size_t length, int flags) override;
 
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                           Event::FileTriggerType trigger, uint32_t events) override;
+  void enableFileEvents(uint32_t events) override;
+
 private:
   void reEnableEventBasedOnIOResult(const Api::IoCallUint64Result& result, uint32_t event);
+
+  // on Windows we use the MSG_PEEK of recv instead of peeking the socket
```

Comment:
on Windows we use the MSG_PEEK of recv -> on Windows we use the MSG_PEEK on recv ?